menu/views.py

```diff
@@ -1,3 +1,31 @@
-from django.shortcuts import render
+import json
 
-# Create your views here.
+from django.views import View
+from django.http import JsonResponse
+
+from menu.models import Category, SubCategory
+
+class CategoryView(View):
+    def get(self, request):
+         try:
+            categories = Category.objects.prefetch_related('subcategory_set').all()
+
+            if categories.exists():   
+                categories = [{ 
+                 'categoryId'              :  category.id,
+                 'category'           :  category.name,
+                 'subCategories' : [
+                     sub_category.name
+                for sub_category in category.subcategory_set.all()]
+            } for category in categories]
+             
+            return JsonResponse({"categories" : categories}, status=200)
+            
+         except ValueError:
+             return JsonResponse({'message':'VALUE_ERROR'}, status=400)
+        
+         except:
+             return JsonResponse({'message':'ERROR'}, status=400)
```

Comment:
```suggestion
         except:
             return JsonResponse({'message':'ERROR'}, status=400)
```
위와 같은 익셉션이 발생하면 무조건 400 및 에러 처리를 하는 것은 제일 하지 말아야 할 처리입니다.
여러분이 익셉션을 사용하는것은 500 에러가 났을 경우, 정확한 에러케이스를 캐치해서 적절한 리턴을 하기 위함인데, 
위와 같이 처리하면 500 에러가 나도 클라이언트에게 잘못을 전가함과 동시에 본인의 정확한 에러도 찾을 수 없는 무책임한 코드 입니다.
앞으로 진행하는 코드에서는 이런 코드는 절대 사용하지 마세요.

Reply:
감사합니다 수정하겠습니다